plugin/videofilterrunnable.cpp

```diff
@@ -0,0 +1,112 @@
+/**
+ * @file videofilterrunnable.cpp
+ * @copyright 2020 Open Mobile PLatform Ltd.
+ * @author Dmitry Butakov d.butakov@omprussia.ru
+ */
+
+#include "videofilterrunnable.h"
+
+#include <QDebug>
+#include <QtDBus/QDBusConnection>
+
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <unistd.h>
+
+VideoFilterRunnable::VideoFilterRunnable(QrFilter *filter) : m_filter(filter)
+{
+    connect(this, &VideoFilterRunnable::stringFound,
+            m_filter, &QrFilter::setResult, Qt::QueuedConnection);
+    // analyze empty frame to establish D-Bus connection
+    analyze(QDBusUnixFileDescriptor(0), 0,
+            QVideoSurfaceFormat(QSize(0,0), QVideoFrame::PixelFormat::Format_BGRA32));
+}
+
+static QString getRandomString()
+{
+   const QString possibleCharacters("ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+                                    "abcdefghijklmnopqrstuvwxyz0123456789");
+   const int randomStringLength = 12;
+
+   QString randomString;
+   for (int i = 0; i < randomStringLength; ++i) {
+       int index = qrand() % possibleCharacters.length();
+       QChar nextChar = possibleCharacters.at(index);
+       randomString.append(nextChar);
+   }
+   return randomString;
+}
+
+QVideoFrame VideoFilterRunnable::run(QVideoFrame *input, const QVideoSurfaceFormat &surfaceFormat,
+                                     QVideoFilterRunnable::RunFlags flags)
+{
+    Q_UNUSED(flags)
+
+    auto in = !input ? QVideoFrame() : *input;
+    if (!m_reply.isFinished()) {
+        return in;
+    }
+
+    if (m_reply.isError()) {
+        qWarning() << "QR Filter error" << m_reply.error().message();
+    } else {
+        QString msg = m_reply.value();
+        if (!msg.isEmpty()) {
+            emit stringFound(msg);
+        }
```

Comment:
Slghtly peculiar this check for result in the analysis of the next frame, but guess it works and is expected to be called shortly.